vassal-app/src/main/java/VASSAL/configure/HotKeyConfigurer.java

```diff
@@ -113,28 +115,7 @@ public void keyReleased(KeyEvent e) {
    * from {@link KeyEvent#getKeyText}
    */
   public static String getString(KeyStroke k) {
-    if (k == null) {
-      return null;
-    }
-
-    String s = KeyEvent.getKeyText(k.getKeyCode());
-    s = s.replace(' ', '_');
-
-    final int mods = k.getModifiers();
-
-    if ((mods & KeyEvent.SHIFT_DOWN_MASK) > 0) {
-      s = Resources.getString("Keys.shift") + " " + s; //$NON-NLS-1$ //$NON-NLS-2$
-    }
-    if ((mods & KeyEvent.CTRL_DOWN_MASK) > 0) {
-      s = Resources.getString("Keys.ctrl") + " " + s; //$NON-NLS-1$ //$NON-NLS-2$
-    }
-    if ((mods & KeyEvent.META_DOWN_MASK) > 0) {
-      s = Resources.getString("Keys.meta") + " " + s; //$NON-NLS-1$ //$NON-NLS-2$
-    }
-    if ((mods & KeyEvent.ALT_DOWN_MASK) > 0) {
-      s = Resources.getString("Keys.alt") + " " + s; //$NON-NLS-1$ //$NON-NLS-2$
-    }
-    return s.toUpperCase();
+    return Chatter.getKeyString(k);    
```

Comment:
Ah! For a very specific reason: because where it was before was inaccessible to module designers to be overridden, and meanwhile Chatter is (or at least will be soon) very easy to override. Since I hate the Java default strings passionately I'm excited to clean them up even more in my modules, but I couldn't resist offering overridability as a gift.

Reply:
What's wrong with them? Is this something we should just standardize instead?

Reply:
Well for example NUMPAD_+ and NUMPAD_- especially the latter, look really bad when rendered in Vassal's fonts and with those underscores inserted. And of course e.g. "CLOSE_BRACKET" in addition to rendering seven miles long is a confusing name to non-Americans. I cleaned up what I think of as the worst offenders, but it seemed like the kind of thing that "belongs in an accessible place" since it deals in interface. The Chatter already has a rare top-level buildFile entry, and already traffics in related areas (text).

Reply:
The fonts can be different, different JDKs use different font rendering engines, they are also different in the same JDK on different platforms.

Reply:
I'd be willing to defend the position that NUMPAD_- looks bad (and confusing) in all fonts and on all platforms :)

Reply:
Where do these strings appear in a module? I'm missing the context here.

Reply:
Ok. Maybe I'm being really dense here, but why can't module designers override `HotKeyConfigurer.getString()`?

Reply:
So the things that are easiest to override are the ones that "have an entry" in buildFile. The easiest of all are the ones that have an entry directly under the main module component -- Chatter is one of these. Meanwhile there are a fair number of high level things that have entries under Map (KeyBufferer, MapCenterer, MenuDisplayer, that kind of stuff -- basically most of VASSAL.build.module.map) and so you can easily override one of those for a particular map. Finally, there is a _different_ override angle for GamePiece traits that "gets in" through the door of the BasicCommandEncoder's entry and overriding that. 

`HotKeyConfigurer` has no such entry so it's hard to override without some kind of major surgery.

Reply:
![image](https://user-images.githubusercontent.com/3742246/88600894-4fe1c500-d03d-11ea-96a7-d465d20ff789.png)

Reply:
Basically in the Custom Class ecology you need to have "a lever" to be able to get at part of the engine and install an override class, and that lever is the buildFile entry. If something is buried down deep w/o a build file entry it's basically untouchable to the module designer -- you can _call into_ those methods, but you can't _override_ them in a way that makes everything else call your override instead of the old method etc in the way you normally can when you're just "writing java".

Custom Class design has its own little jujitsu you have to learn -- in many ways it's MUCH easier to do something directly in Vassal than to accomplish it in a custom class.

Reply:
(1) I keep thinking that there has to be a better place for this than in `Chatter`, and (2) I'm not convinced that we need multiple ways to print the names of keys.

So, could we take this one change out of this PR and have it under its own issue---since it's orthogonal to fixing the Mac issue this PR is for? I'm not saying no to this, I just would like it not to hold up the Mac fixes further.

Reply:
Yes, the problem is not so much over-riding a particular class, but getting your over-ridden class to be used instead of the original Vassal class. There are plenty of classes that would be nice to over-ride, but you can't get a 'handle' on them.

Having written that, I realised that it should be possible to over-ride any class if the ClassLoader looked in the module first before looking to Vassal for a class. Not sure what the order is at the moment.

Reply:
That will work only if the class isn't loaded before the module is.

Reply:
I can certainly move it out, but I do want to correct something you just said that seems wrong. We don't have "multiple ways" to print the name of keys, this method (whichever class it ends up in) is literally the one-central-place-for-all-VASSAL to print the name of keys. I completely agree that we should have exactly one place -- and this method is it (again, whatever class it's in). 

And the reason I think module designers will want to override it is that I know I did, except that since there wasn't a "handle" for it as Brent puts it, I had to do a godawful override of MenuDisplayer that involved intercepting three different code-paths, and all that did was change what came up in Context Menus, there was no direct way to get at what's in Toolbar items and stuff.

Reply:
I only picked Chatter because out of all the "easy to override things" it was the one most closely related to the keyboard. Admittedly a tangentially different flavor of keyboard.

Reply:
No, I'm not saying we have multiple ways _now_, but your goal seems to be to let module designers have multiple ways to display key names. Is there a reason we can't devise one good way so they don't need to?

Reply:
We can certainly have a _go_ at it (and should always be trying to improve that), but then there are module designers "in other lands" who will want to fix e.g. non-English things that we aren't even thinking about, or the English ones fixing other things like mis-naming the [ ] symbols using the words non-Americans refer to ( ) symbols by, but that we haven't thought of yet. And somebody who's going to want the names in their module displayed in all lower case or all-whatever-case. Etc. Given there's no performance/efficiency problem that I'm aware of, it's a why NOT provide as many hooks as possible, and provide them in ways that make custom classes easier rather than harder to implement/maintain.

Reply:
This sounds like something that could be moved into it's own top-level Buildable with a Buildfile entry. That would make it easily over-rideable. It wouldn't need to be Configurable (e.g. Like the CommandEncoder buildable isn't), but it could be if you can work out a way to make some of the flexibility you want Configurable. You could design the new Buildable to be easily sub-classed.

Reply:
Okay, back at my actual keyboard and have pushed a version of this w/ keys moved back to HotKeyConfigurer for now.

Reply:
@BrentEaston That's worth exploring. If we added that, what would become of module buildFiles that didn't include that line?

Reply:
That's all handled by BasicModule. When creating a new module, the components defined in buildDefaultCompenents() are added to the module automatically (List 1).  
    addComponent(BasicCommandEncoder.class);
    addComponent(Documentation.class);
    addComponent(PlayerRoster.class);
    addComponent(GlobalOptions.class);
    addComponent(Map.class);
    addComponent(GamePieceImageDefinitions.class);
    addComponent(GlobalProperties.class);
    addComponent(PrototypesContainer.class);
    addComponent(PieceWindow.class);
    addComponent(Chatter.class);
    addComponent(Language.class);
These are the default top-level components that every new Module has.

When loading an existing module, the code at BasicModule:143 runs and adds 3 specific components if they don't exist. These are the top-level components that a running module can't do without. (List 2)
      ensureComponent(GamePieceImageDefinitions.class);
      ensureComponent(GlobalProperties.class);
      ensureComponent(Language.class);

Why these particular classes, I couldn't tell you as there are components in the first list that would stop a module running if they where missing from an existing module (CommandEncoder for instance).

This is the same place where we can potentially fix the 'Oops, I accidentally deleted my PlayerRoster, how do I get it back' issue by adding all of the classes in list 1 to list 2.